src/answers-search-bar.js

```diff
@@ -0,0 +1,519 @@
+/** @module */
+
+import Core from './core/core';
+
+import {
+  Renderers,
+  DOM
+} from './ui/index';
+
+import ErrorReporter from './core/errors/errorreporter';
+import { AnalyticsReporter } from './core';
+import Storage from './core/storage/storage';
+import { AnswersComponentError } from './core/errors/errors';
+import AnalyticsEvent from './core/analytics/analyticsevent';
+import StorageKeys from './core/storage/storagekeys';
+import QueryTriggers from './core/models/querytriggers';
+import SearchConfig from './core/models/searchconfig';
+import ComponentManager from './ui/components/componentmanager';
+import VerticalPagesConfig from './core/models/verticalpagesconfig';
+import { SANDBOX, PRODUCTION, LOCALE, QUERY_SOURCE } from './core/constants';
+import { isValidContext } from './core/utils/apicontext';
+import { urlWithoutQueryParamsAndHash } from './core/utils/urlutils';
+import Filter from './core/models/filter';
+import SearchComponent from './ui/components/search/searchcomponent';
+import QueryUpdateListener from './core/statelisteners/queryupdatelistener';
+import { SEARCH_BAR_COMPONENTS_REGISTRY } from './ui/components/search-bar-only-registry';
+
+/** @typedef {import('./core/services/errorreporterservice').default} ErrorReporterService */
+/** @typedef {import('./core/services/analyticsreporterservice').default} AnalyticsReporterService */
+
+/**
+ * @typedef Services
+ * @property {ErrorReporterService} errorReporterService
+ */
+
+const DEFAULTS = {
+  locale: LOCALE,
+  querySource: QUERY_SOURCE
+};
+
+/**
+ * The class that powers the stand-alone SearchBar integration. Note that this class
+ * follows the singleton pattern.
+ */
+class AnswersSearchBar {
+  constructor () {
+    if (!AnswersSearchBar.setInstance(this)) {
+      return AnswersSearchBar.getInstance();
+    }
+
+    /**
+     * A reference of the renderer to use for the components
+     * This is provided during initialization.
+     * @type {Renderer}
+     */
+    this.renderer = new Renderers.Handlebars();
+
+    /**
+     * A local reference to the component manager
+     * @type {ComponentManager}
+     */
+    this.components = new ComponentManager(SEARCH_BAR_COMPONENTS_REGISTRY);
+
+    /**
+     * A local reference to the core api
+     * @type {Core}
+     */
+    this.core = null;
+
+    /**
+     * A callback function to invoke once the library is ready.
+     * Typically fired after templates are fetched from server for rendering.
+     */
+    this._onReady = function () {};
+
+    /**
+     * @type {boolean}
+     * @private
+     */
+    this._eligibleForAnalytics = false;
+
+    /**
+     * @type {Services}
+     * @private
+     */
+    this._services = null;
+
+    /**
+     * @type {AnalyticsReporterService}
+     * @private
+     */
+    this._analyticsReporterService = null;
+  }
+
+  static setInstance (instance) {
```

Comment:
nit: can you add some jsdoc explaining the return value? Specifically that true is returned only if the instance is able to be set